File: C++/BubbleSort2.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

void bubbleSort(int s[], int const N) {
	int temp;
	for (int i=0; i<N-1; i++)
		for(int j=0; j<N-1; j++)
			if (s[j]>s[j+1]) {
				temp = s[j];
				s[j] = s[j+1];
				s[j+1] = temp;
			}
}
	
int main() {
	int arraySize;
	srand(time(NULL));

	cout << "Input array size: ";
	cin >> arraySize;
    int* s = new int[arraySize];
	int const N=arraySize;
    for (int i = 0; i < N; i++) {
		s[i] = (rand() % 10000);
	}
	for (int k=0; k<N-1; k++) {
		cout << s[k] << " ";
	}
	cout << "\n";
	
	bubbleSort(s, N);
	for (int k=0; k<N; k++) {
		cout << s[k] << " ";
	}
	cout << "\n";
	return 0;
}

File: Node/BubbleSort2.js
// ====== 共通の入出力機能（変更しない）======
import { output, input } from "./io.js";
// ==========================================

function rand() { return Math.floor(Math.random() * 2147483648); }

function bubbleSort(s) {
  const N = s.length;
  for (let i = 0; i < N - 1; i++)
    for (let j = 0; j < N - 1; j++)
      if (s[j] > s[j + 1]) {
        [s[j], s[j + 1]] = [s[j + 1], s[j]];
      }
}

function main() {
  const arraySize = parseInt(input("Input array size: "));
  const s = [];     // JSの配列は自動拡張されるため大きさの指定は不要
  const N = arraySize;
  for (let i = 0; i < N; i++) {
    s[i] = rand() % 10000;
  }
  for (const v of s) {
    output(v + " ");
  }
  output("\n");

  bubbleSort(s);
  for (const v of s) {
    output(v + " ");
  }
  output("\n");
}

main();
